gradio/blocks.py

```diff
@@ -1173,14 +1173,80 @@ def getLayout(block):
             if serializer:
                 assert isinstance(block, serializing.Serializable)
                 block_config["serializer"] = serializer
-                block_config["info"] = {
-                    "input": list(block.input_api_info()),  # type: ignore
-                    "output": list(block.output_api_info()),  # type: ignore
-                }
+                block_config["info"] = block.api_info()
             config["components"].append(block_config)
         config["dependencies"] = self.dependencies
         return config
 
+    def get_api_info(self, raw=False):
```

Comment:
@abidlabs If we want the api fetcher to call `gr.load`, then we need `gr.load` to work for apps not hosted in spaces. Is that what we want to do? There was some hesitation about that before: https://huggingface.slack.com/archives/C02SPHC1KD1/p1667487172025779?thread_ts=1667401251.893519&cid=C02SPHC1KD1

Reply:
@abidlabs Would it be possible to refactor the existing logic so that the api info can be fetched entirely from the config without having to instantiate a client or call `gr.external.from_spaces_blocks()`.

Cause I just realized that, as things stand now, the api fetcher would not work for apps running locally, but a workaround could be to send the config over to the server. That would work for local, public, private apps.

Reply:
Maybe I'm misunderstanding, but even if I refactor the existing logic so that there's a way to get the api info without instantiating a client, that would only apply to new versions of Gradio. In fact, such a method does exist -- `Blocks.get_api_info()`. But the API fetcher needs to be able to go from an older version of Gradio -> API Info, and the only way that would be possible is to load the Space

Reply:
Oh nvm I understand what you mean now

Reply:
Okay yeah I think I can quicklyrewrite `get_api_info` to go from config to API info

Reply:
On main, my understanding was that the logic for `Client.view_api` works for all versions of gradio? 

I'm suggesting we refactor that so that it's a standalone function/static method.

Let me know if I'm crazy lol

Reply:
Lol sweet thanks @abidlabs

Reply:
Okay fixed. So now `get_api_info` is a standalone method in `gradio.blocks` that goes from config to api info. Lmk if that works!

Reply:
For testing purposes, you could even just copy paste the `gradio.blocks.get_api_info ()` function into your Space (rather than having to do a release of the library to use in the Space)

Reply:
It’s all good - I’m uploading the wheel to s3 so we’re covered

Reply:
I believe this checks for  `if not component.get("serializer"):` in the function make it so that the function will not return anything for older spaces

<img width="616" alt="image" src="https://user-images.githubusercontent.com/41651716/230695866-ffa7d112-ec04-4c67-ab8f-9b17975dd1a4.png">

Reply:
Ah good point, will fix tomorrow

Reply:
Should be working now: 

<img width="732" alt="image" src="https://user-images.githubusercontent.com/1778297/230727114-d1fbf335-1b38-
4324-a3c0-6e73911933b3.png">

Reply:
Awesome thank you! Just pushed up the api fetching logic

Reply:
Sweet! Will test it out